api/job_resources.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type JobResources struct {
+	client *Client
+}
+
+// JobResources returns a handle on the JobResources endpoints
+func (c *Client) JobResources() *JobResources {
+	return &JobResources{client: c}
+}
+
+// List returns a list of all resources for a particular context. If a
+// context is not specified, matches for all contezts are returned.
+func (j *JobResources) List(prefix string, context string) (*structs.ResourceListResponse, error) {
```

Comment:
Don't want to take or return anything from `structs` in this package. Create a copy of the ListResponse here.